cloud/azure/management/costs/mode/costusage.pm

```diff
@@ -0,0 +1,214 @@
+    $self->{result_values}->{budget} = $options{new_datas}->{$self->{instance} . '_budget'};
+    $self->{result_values}->{prct_cost} = ($self->{result_values}->{budget} != 0) ? $self->{result_values}->{cost} * 100 / $self->{result_values}->{budget} : 0;
+    return 0;
+}
+
+sub set_counters {
+    my ($self, %options) = @_;
+
+    $self->{maps_counters_type} = [
+        { name => 'cost', type => 0 }
+	];
+
+    $self->{maps_counters}->{cost} = [
+        { label => 'cost', set => {
+	    key_values => [ { name => 'cost' }, { name => 'budget' } ],
+                closure_custom_calc => $self->can('custom_cost_calc'),
+                closure_custom_output => $self->can('custom_cost_output'),
+                closure_custom_threshold_check => $self->can('custom_cost_threshold'),
+                closure_custom_perfdata => $self->can('custom_cost_perfdata')
+	  }
+        },
+	];
+}
+
+sub new {
+    my ($class, %options) = @_;
+    my $self = $class->SUPER::new(package => __PACKAGE__, %options, force_new_perfdata => 1);
+    bless $self, $class;
+
+    $options{options}->add_options(arguments => {
+	"budget-name:s"     => { name => 'budget_name' },
+	"resource-group:s"  => { name => 'resource_group' },
+	"lookup-days:s"     => { name => 'lookup_days', default => 30 },
+	"units:s"           => { name => 'units', default => '%' },
+	"timeout:s"           => { name => 'timeout', default => '60' },
+    });   
+
+    return $self;
+}
+
+sub check_options {
+    my ($self, %options) = @_;
+    $self->SUPER::check_options(%options);
+
+    if (!defined($self->{option_results}->{budget_name}) || $self->{option_results}->{budget_name} eq '') {
+	$self->{output}->add_option_msg(short_msg => "Need to specify --budget-name option");
+	$self->{output}->option_exit();
+    }
+
+    $self->{lookup_days} = $self->{option_results}->{lookup_days};
+}
+
+sub manage_selection {
+    my ($self, %options) = @_;
+
+    my $budget = $options{custom}->azure_get_budget(
+	resource_group => $self->{option_results}->{resource_group},
+	budget_name => $self->{option_results}->{budget_name}
+	);
+
+    my $usage_start = DateTime->now()->add(days => - $self->{option_results}->{lookup_days} + 1);
+    my $usage_end = DateTime->now();
+    my $costs = $options{custom}->azure_get_usagedetails(
+	resource_group => $self->{option_results}->{resource_group},
+	usage_start    => $usage_start,
+	usage_end      => $usage_end
+	);
+    my $cost = 0;
+    for (my $i = 0; $costs->[$i]; $i++) {
+	$cost += $costs->[$i]->{properties}->{cost};
+    }
+
+    if (!$budget) {
+	$self->{output}->add_option_msg(short_msg => "No " . $self->{option_results}->{budget_name} . " found (or missing permissions)");
+	$self->{output}->option_exit();
+    }
+    if (!$costs || $cost < 0.01) {
+	$self->{output}->add_option_msg(short_msg => "Null or < 0.01 " . $budget->{properties}->{currentSpend}->{unit} . "cost found on the specified scope.");
+	$self->{output}->option_exit();
+    }
```

Comment:
@matoy , would it be a problem to force à 0 here? Could it hide some errors (mismatch resourcegroup vs budget)?